docs/rfc/0001-framework-migration.md

```diff
@@ -0,0 +1,103 @@
+* test framework should be unified as `Ava`. Unit test cases should be as free of IO and side effects as possible, the parallel running mechanism of AVA will force us to write more efficient test codes
+
+Switching the test framework to `Ava` will bring some refactoring work:
+
+```js
+// ava
+import test from 'ava';
+import * as fs from 'fs-extra';
+import * as sinon from 'sinon';
+import { shuffle } from './public';
+
+test('array shuffle', (t) => {
+  const array = [ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 ];
+  shuffle(array);
+  t.notDeepEqual(array, [ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 ]);
+  t.deepEqual(array.sort(), [ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 ]);
+});
+
+// serial hook
+test.serial.afterEach(() => {
+  sinon.restore();
+});
+
+// serial case
+test.serial('test a', (t) => {
+  const fsReadJsonMock = sinon.stub(fs, 'readJson').resolves({});
+  t.deepEqual(await fs.readJson('mockFileName.json'), {});
+});
+
+test.serial('test b', (t) => {
+  const fsReadJsonMock = sinon.stub(fs, 'readJson').resolves(undefined);
+  t.is(await fs.readJson('mockFileName.json'), undefined);
+});
+
+test.todo('some todo cases');
+```
+
+```js
+// jasmine
+import { shuffle } from './public';
+
+describe('public utils', () => {
+  it('test if the array is shuffled', () => {
+    const array = [ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 ];
+    shuffle(array);
+    expect(array).not.toEqual([ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 ]);
+    expect(array.sort()).toEqual([ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 ]);
+  });
+}
+```
+
+## Integration test
+
+The integration test of 1.x depends on the workflow configuration file. In a strict sense, this is not a rigorous integration testing method and hard to maintain. In addition, we do not assert any of the running processes except exit code. This situation will be improved in 2.0 as follows:
+
+* define integration test cases
+* use or develop an integration test framework suitable for Pipbook
+* extend the run environment
+
+# Framework migration for Daemon
+
+The framework of daemon will be moved from the original framework to `loopback 4` for the following reasons:
+
+* automatic code generation capability
+* framework maturity: the version of typescript cannot be configured in the original framework, resulting in the overall size of the pipbook package unable to be trimmed, and the test framework is constrained and cannot be switched
+* i18n
+
+# Framework migration for Pipboard
+
+The pipboard UI library will be migrated from the ice.js to the [ant.design](https://ant.design/) for the following reasons:
+- The icejs has no English documentation, so it is not easy to contribute
+- Compared with the familiarity of community contributors, the ant.design library is higher than icejs
+- Reduce the complexity of the framework to make it easy to maintain. Ice.js contains some complex functions that are unnecessary for pipboard, such as MPA (Multi-page application), SSR (Server-side rendering), Permission routing, etc.
```

Comment:
Ditto, Ice.js, ice.js and icejs, we need to choose one.

Reply:
:) Use icejs, its [documentation](https://ice.work/docs/guide/migrate) is written like this